src/bin/activate.rs

```diff
@@ -425,7 +431,7 @@ pub async fn activate(
             info!("Activation succeeded!");
         }
 
-        if magic_rollback {
+        if magic_rollback && !boot {
```

Comment:
I think there should be a warning saying that magic_rollback doesn't work with dry-activate or boot